### ISSUE TITLE
Fix sc.animation

### ISSUE THREAD

**Issue by cliffckerr** [sciris/sciris#291]
- [x] Fix unpassed kwargs in `save()`
- [x] Add `anim.ffmpeg()`
- [x] Fix bug, whatever it is -- only saves last frame and takes 20+ GB of memory!!

**Comment by pausz** [comment]
First pass, using the example in sc.animation()

```python
import sciris as sc
import pylab as pl 

def animation():

    anim = sc.animation()

    pl.figure()
    pl.seed()

    repeats = 21
    colors = sc.vectocolor(repeats, cmap='turbo')
    for i in range(repeats):
        scale = 1/pl.sqrt(i+1)
        x = scale*pl.randn(10)
        y = scale*pl.randn(10)
        label = str(i) if not(i%5) else None
        pl.scatter(x, y, c=[colors[i]], label=label)
        pl.title(f'Scale = 1/√{i}')
        pl.legend()
        sc.boxoff('all')
        anim.addframe()

    anim.save('dots.mp4')


sc.mprofile(animation)
```

Mem profiling output:
```
Line #    Mem usage    Increment  Occurrences   Line Contents
=============================================================
     6     73.1 MiB     73.1 MiB           1   def animation():
     7                                         
     8     73.1 MiB      0.0 MiB           1       anim = sc.animation()
     9                                         
    10     81.0 MiB      7.9 MiB           1       pl.figure()
    11     81.0 MiB      0.0 MiB           1       pl.seed()
    12                                         
    13     81.0 MiB      0.0 MiB           1       repeats = 21
    14     81.0 MiB      0.0 MiB           1       colors = sc.vectocolor(repeats, cmap='turbo')
    15    103.8 MiB      0.0 MiB          22       for i in range(repeats):
    16    103.8 MiB      0.0 MiB          21           scale = 1/pl.sqrt(i+1)
    17    103.8 MiB      0.0 MiB          21           x = scale*pl.randn(10)
    18    103.8 MiB      0.0 MiB          21           y = scale*pl.randn(10)
    19    103.8 MiB      0.0 MiB          21           label = str(i) if not(i%5) else None
    20    103.8 MiB      1.3 MiB          21           pl.scatter(x, y, c=[colors[i]], label=label)
    21    103.8 MiB      0.0 MiB          21           pl.title(f'Scale = 1/√{i}')
    22    103.8 MiB      0.3 MiB          21           pl.legend()
    23    103.8 MiB     -0.0 MiB          21           sc.boxoff('all')
    24    103.8 MiB     21.2 MiB          21           anim.addframe()
    25                                         
    26    514.9 MiB    411.1 MiB           1       anim.save('dots.mp4')

```

OS: Ubuntu 20.04
ffmpeg: ffmpeg version 4.2.4-1ubuntu0.1, built with gcc 9 (Ubuntu 9.3.0-10ubuntu2)
python: 3.9.12

**Comment by cliffckerr** [comment]
Closed by #348
